src/source/PeerConnection/SessionDescription.c

```diff
@@ -436,20 +437,47 @@ STATUS populateSingleMediaSection(PKvsPeerConnection pKvsPeerConnection, PKvsRtp
     SPRINTF(pSdpMediaDescription->sdpAttributes[attributeCount].attributeValue, "%d", mediaSectionId);
     attributeCount++;
 
-    switch (pKvsRtpTransceiver->transceiver.direction) {
-        case RTC_RTP_TRANSCEIVER_DIRECTION_SENDRECV:
-            STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "sendrecv");
-            break;
-        case RTC_RTP_TRANSCEIVER_DIRECTION_SENDONLY:
-            STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "sendonly");
-            break;
-        case RTC_RTP_TRANSCEIVER_DIRECTION_RECVONLY:
-            STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "recvonly");
-            break;
-        default:
-            // https://www.w3.org/TR/webrtc/#dom-rtcrtptransceiverdirection
-            DLOGW("Incorrect/no transceiver direction set...this attribute will be set to inactive");
-            STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "inactive");
+
+    if ( pKvsPeerConnection->isOffer ) {
+        switch (pKvsRtpTransceiver->transceiver.direction) {
+            case RTC_RTP_TRANSCEIVER_DIRECTION_SENDRECV:
+                STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "sendrecv");
+                break;
+            case RTC_RTP_TRANSCEIVER_DIRECTION_SENDONLY:
+                STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "sendonly");
+                break;
+            case RTC_RTP_TRANSCEIVER_DIRECTION_RECVONLY:
+                STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "recvonly");
+                break;
+            default:
+                // https://www.w3.org/TR/webrtc/#dom-rtcrtptransceiverdirection
+                DLOGW("Incorrect/no transceiver direction set...this attribute will be set to inactive");
+                STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "inactive");
+        }
+    } else {
+        for ( int i = 0; i < pRemoteSessionDescription->mediaCount; i++ ) {
+
+            PSdpMediaDescription pSdpMediaDescriptionRemote = &pRemoteSessionDescription->mediaDescriptions[i];
+
+            UINT32 remoteAttributeCount = pSdpMediaDescriptionRemote->mediaAttributesCount;
+
+            for (int i = 0; i < remoteAttributeCount; i++) {
+                if (STRCMP(pSdpMediaDescriptionRemote->sdpAttributes[i].attributeName, "sendrecv") == 0) {
+                    STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "sendrecv");
+                    break;
```

Comment:
I generally don't recommend code flow modifier keywords in complex loops/nested loops.

Instead, in this case, have a boolean - something like iterate = TRUE; then, when you need to exit the loop, set it's value to FALSE which will then be checked in the loop terminator condition.

Next, make all these if to if/else ifs